ancpbids/pybids_compat.py

```diff
@@ -38,9 +38,17 @@ def _to_any_expr(self, value, ctor):
         # else just return using the constructor function
         return ctor(value)
 
-    def __getattr__(self, key, **kwargs):
-        k = key if not key.startswith("get_") else key[4:]
-        return partial(self.get, return_type='id', target=k, **kwargs)
+    def __getattr__(self, key):
+        # replace arbitrary get functions with calls to get
+        if key.startswith("get_"):
+            return partial(self.get, "id", key[4:])
+
+        # look for BL attributes as attributes of the BL.dataset
```

Comment:
is this behavior also present in pybids? relaying to self.dataset looks arbitrarily chosen or what is the rationale behind this?

Reply:
In the case of `pybids`, many attributes (e.g., `files`) are properties of the layout and are directly accessible. That is not the case with your implementation. This getter function does not overwrite any attributes of the `ancpbids` layout, but does provide similar convenience when people are trying to extract `dataset`-level values.